File: Python3/515. Find Largest Value in Each Tree Row.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def largestValues(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        if not root: return [] # 68ms
        currlevel = [root]
        nextlevel = []
        ans = []
        while len(currlevel) > 0:
            maxvalue = currlevel[0].val
            while len(currlevel) > 0:
                node = currlevel.pop()
                if node.val > maxvalue:
                    maxvalue = node.val
                if node.left: nextlevel.append(node.left)
                if node.right: nextlevel.append(node.right)
            ans.append(maxvalue)
            currlevel = nextlevel
            nextlevel = []
        return ans

File: Golang/515. Find Largest Value in Each Tree Row.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func largestValues(root *TreeNode) []int { // 20ms
    if root == nil {
        return []int{}
    }
    currlevel := []*TreeNode{root}
    nextlevel := []*TreeNode{}
    ans := []int{}
    for len(currlevel) > 0 {
        maxvalue := currlevel[0].Val
        for len(currlevel) > 0 {
            node := currlevel[0]
            currlevel = currlevel[1:]
            if node.Val > maxvalue {
                maxvalue = node.Val
            }
            if node.Left != nil {
                nextlevel = append(nextlevel, node.Left)
            }
            if node.Right != nil {
                nextlevel = append(nextlevel, node.Right)
            }
        }
        ans = append(ans, maxvalue)
        currlevel = nextlevel
        nextlevel = []*TreeNode{}
    }
    return ans
}
